lock-api-objects/src/main/java/com/palantir/lock/watch/LockEvent.java

```diff
@@ -52,6 +66,63 @@ public <T> T accept(Visitor<T> visitor) {
         return visitor.visit(this);
     }
 
+    @JsonCreator
+    public static LockEvent create(
+            @JsonProperty("sequence") long sequence,
+            @JsonProperty("lockDescriptors") List<LockDescriptor> lockDescriptors,
+            @JsonProperty("lockToken") LockToken lockToken,
+            @JsonProperty("metadata") Optional<WireLockRequestMetadata> optWireMetadata) {
+        Optional<LockRequestMetadata> metadata = optWireMetadata.map(wireMetadata -> {
+            KeyListChecksum checksum = KeyListChecksum.of(
+                    ChecksumType.valueOf(wireMetadata.lockListChecksum().typeId()),
+                    wireMetadata.lockListChecksum().value());
+            IndexEncodingResult<LockDescriptor, ChangeMetadata> indexEncodingResult =
+                    IndexEncodingResult.<LockDescriptor, ChangeMetadata>builder()
+                            .keyList(lockDescriptors)
+                            .indexToValue(wireMetadata.indexToChangeMetadata())
+                            .keyListChecksum(checksum)
+                            .build();
+            Map<LockDescriptor, ChangeMetadata> lockDescriptorToChangeMetadata =
+                    IndexEncodingUtils.decode(indexEncodingResult, Function.identity());
+            return LockRequestMetadata.of(lockDescriptorToChangeMetadata);
+        });
+        return ImmutableLockEvent.builder()
+                .sequence(sequence)
+                .lockDescriptors(ImmutableSet.copyOf(lockDescriptors))
+                .lockToken(lockToken)
+                .metadata(metadata)
+                .build();
+    }
+
+    @Unsafe
+    @JsonValue
+    public WireLockEvent toWireLockEvent() {
```

Comment:
There are alternatives to using a custom immutable as the return type here, but they are all less compelling imo. The hacky way would be to return a `Map<String, Object>`